docs/resource_management.md

```diff
@@ -0,0 +1,68 @@
+# Resource management
+
+This document describes the fundamentals of resource management in permissions-api.
+
+## Resource types
+
+Every resource in permissions-api has a type that defines the actions and relationships that can be scoped to that resource. By default, the only two resource types that exist in permissions-api are _role_ and _tenant_, which define sets of allowed actions and operational context respectively and are necessary for the service to function.
+
+## Resource lifecycle events
+
+permissions-api consumes resource lifecycle events over [NATS][nats]. This section describes the subjects and message formats the service expects.
+
+[nats]: https://nats.io
+
+### Message format
+
+permission-api expects lifecycle event messages in the format described in [`go.infratographer.com/x/pubsubx`][pubsubx], and interprets the message fields as follows:
+
+* `subject_urn`: The URN of the resource the event is about
+* `event_type`: The type of lifecyle event for the resouce. Must match one of the defined lifecycle events
+* `additional_subjects`: Resources related to the resource. These map to a relationship with both a relation and resource type that match the resource type in the URN
```

Comment:
I had a discussion about this today, this will also be in `SubjectFields` as `tenant_id` (ie the db column name) which may be a better place to reference the value - it will not be the URN though, just the id.